### DISCUSSION TITLE
How to generate tokens with different expiration time?

### DISCUSSION THREAD

**Discussion by mariojsnunes** [OrchardCMS/OrchardCore#10716] [Q&A] [answered]
I have an integreation with a service that requires a long-lived Bearer token as they do not implement a refresh tokens or client credentials yet.
I only want to override the expiration value for this specific API, all the rest should stay the same.
I've found a few topics about OpenIddict like this one https://stackoverflow.com/questions/42442109/is-there-a-way-to-have-different-ticket-expiry-lengths-in-openiddict where you can override the global value. Am I suposed to implement one? Are there any samples?

**Comment by Skrypt** [comment, marked as answer]
Seems like it can be set Globally only.

**Reply by mariojsnunes** [reply]
thanks

**Comment by hyzx86** [comment]
FYI.

https://github.com/OrchardCMS/OrchardCore/discussions/7036
